### DISCUSSION TITLE
`--metadata-file` yaml on xelatex and debian - how to debug?

### DISCUSSION THREAD

**Discussion by brunocek** [jgm/pandoc#9550] [Q&A]
Please guidance and help.

How can I debug a pdf generation from a latex template (using xelatex)? I am having trouble bringing metadata from yaml files, is there a way to see log, aux, out files when the pdf is generated? Or intermediate files?

I fear the new environment I am running this is not good enough be it for the python install or for packages, and I don't know how to check these.

---
I migrated from a debian test to a debian-11 (yes, the newer machine has an older version of debian). On the old machine I was able to generate a pdf from a tex template using xelatex, and now it is not working anymore.

I do "import" three yaml files with `--metadata-file`, and I use the data from the yamls. But now, these fields are not rendering anymore.

I use the imported data in loops using `$ for( field ) $`, where `field` is present in the yaml. And I also use: `\newcommand{\field}{ $ field $ }`; and `\def\field#1{\def\@field{#1}}` when I use fields outside of loops.

I am sorry I am not good at latex.

This was also asked here: https://tex.stackexchange.com/questions/712537/pandoc-metadata-file-yaml-on-xelatex-and-debian-how-to-debug

I suspect the problem is that I might not have installed some package. Here follows what I have on the new debian-11 machine.

      - texlive-base
      - texlive-fonts-extra 
      - texlive
      - fonts-texgyre
      - fonts-mathjax
      - texlive-xetex
      - texlive-publishers
      - texlive-plain-generic
      - texlive-pictures
      - texlive-latex-recommended
      - texlive-latex-extra
      - texlive-latex-base
      - texlive-fonts-recommended
      - texlive-binaries
      - python3-pygments
      - python3-jinja2
      - python3-docutils
      - preview-latex-style
      - pandoc-data
      - pandoc
      - libgtksourceview-4-common
      - ispell
      - dvisvgm
      - docutils-common
      - cm-super
      - cm-super-minimal
      - wget

**Comment by jgm** [comment]
>  I am having trouble bringing metadata from yaml files, is there a way to see log, aux, out files when the pdf is generated? Or intermediate files?

Using `--verbose` will give you the intermediate file and other info to stdout.

To see aux, etc., you can use `pandoc -s -t latex` to produce a standalone latex file and run xelatex on it yourself.  Or use `--pdf-engine=latexmk --pdf-engine-opt=-outdir=build` to use latexmk for the build and use `build` as a persistent subdirectory which will contain the intermediate files.
